modules/core/include/opencv2/core/mat.hpp

```diff
@@ -1717,6 +1720,12 @@ class CV_EXPORTS Mat
      */
     size_t total() const;
 
+    /** @brief Returns the total number of array elements.
+
+     The method returns the number of elements within a certain sub-array slice with startDim <= dim < endDim
+     */
+    size_t total(int startDim, int endDim=INT_MAX) const;
```

Comment:
This functionality looks misnamed. Probably `sliceSize` would be good.
Also my suggestion is to move this out of `cv::Mat` class to prevent exploding of cv::Mat with rarely used/specific functionality.